mockingbird/src/commonMain/kotlin/com/careem/mockingbird/test/Capture.kt

```diff
@@ -47,18 +49,59 @@ public fun <T> capture(list: CapturedList<T>): CapturedMatcher<T> {
  */
 
 public class Slot<T> : Captureable {
+
+    private val genericSlot : GenereicSlot<T> = initializeSlot<T>()
+
+    public val captured: T?
+        get() {
+            return genericSlot.value
+        }
+
+    @Suppress("UNCHECKED_CAST")
+    override fun storeCapturedValue(value: Any?) {
+        genericSlot.storeCapturedValue(value)
+    }
+}
+
+private fun <T> initializeSlot(): GenereicSlot<T> {
+    return when (MockingBird.mode) {
+        TestMode.MULTI_THREAD -> MultiThreadSlot()
+        TestMode.LOCAL_THREAD -> LocalThreadSlot()
+    }
+}
+
+private interface GenereicSlot<T> : Captureable {
+    val value: T?
+}
+
+private class MultiThreadSlot<T> : GenereicSlot<T> {
+
     private val _captured: AtomicRef<T?> = atomic(null)
-    public var captured: T?
+
+    @Suppress("UNCHECKED_CAST")
+    override fun storeCapturedValue(value: Any?) {
+        _captured.value = value as T
+    }
+
+    override val value: T?
+        get() = _captured.value
+}
+
+private class LocalThreadSlot<T> : GenereicSlot<T> {
```

Comment:
Nit: `SingleThreadSlot`?

Reply:
I honestly prefer threadLocal is more align with the threadLocal concept

Reply:
@MarcoSignoretto technically "thread local" is something else though, this slot implementation doesn't have per-thread storage, does it?